users/tests/tests.py

```diff
@@ -0,0 +1,43 @@
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+from django.test import TestCase, RequestFactory
+
+from users.forms import RegistrationForm
+from users.views import register
+
+
+class RegistrationPageUnitTests(TestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+
+    def test_method_get(self):
+        request = self.factory.get(reverse('register'))
+        response = register(request)
+        self.assertEqual(response.status_code, 200)
+
+    def test_method_post(self):
+        request = self.factory.post(reverse('register'))
+        response = register(request)
+        self.assertEqual(response.status_code, 200)
+
+    def test_form_is_valid(self):
+        form_data = {'username': "testUser", 'password1': "pass1234", 'password2': "pass1234"}
+        form = RegistrationForm(data=form_data)
+        self.assertTrue(form.is_valid())
+
+    def test_form_is_invalid(self):
+        form_data = {'username': "testUser", 'password1': "pass1234", 'password2': "pass123"}
+        form = RegistrationForm(data=form_data)
+        self.assertFalse(form.is_valid())
+
+
+class RegistrationPageIntegrationTests(TestCase):
+    def test_register_successful(self):
+        response = self.client.post(reverse('register'), {'username': 'user_1', 'password1': 'pass1234', 'password2': 'pass1234'})
+        self.assertEqual(response.content, "You successfully created a new account!")
+
+    def test_register_unsuccessful(self):
+        self.user = User.objects.create_user("user_2")
+        self.user.save()
+        response = self.client.post(reverse('register'), {'username': 'user_2', 'password1': 'pass1234', 'password2': 'pass1234'})
+        self.assertTrue("A user with that username already exists." in response.content)
```

Comment:
I think you can use https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertIn